### PR TITLE
Early JFR framework

### PR DESCRIPTION
**Pull request by gacholio**
- thread local buffer management
- execution sample event

### CODE REVIEW

File: runtime/vm/jfr.cpp [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+ */
+static bool
+flushBufferToGlobal(J9VMThread *currentThread, J9VMThread *flushThread)
+{
+#if defined(DEBUG)
+	PORT_ACCESS_FROM_VMC(currentThread);
+	j9tty_printf(PORTLIB, "\n!!! flushing %p of size %p\n", flushThread, flushThread->jfrData.bufferSize - flushThread->jfrData.bufferRemaining);
+#endif /* defined(DEBUG) */
+
+	// TODO: flush to global
```
Comment (tajila) [inline runtime/vm/jfr.cpp:64]: will this include synchronization to account for simultaneous `reserveBuffer` failures?

File: runtime/vm/jfr.cpp [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+		UDATA eventSize = sizeof(J9JFRExecutionSample) + stackTraceBytes;
+		J9JFRExecutionSample *sample = (J9JFRExecutionSample*)reserveBuffer(currentThread, eventSize);
+		if (NULL != sample) {
+			initializeEventFields(currentThread, (J9JFREvent*)sample, J9JFR_EVENT_TYPE_EXECUTION_SAMPLE, J9JFR_THREAD_STATE_RUNNING);
+			sample->stackTraceSize = framesWalked;
+			memcpy(J9JFREXECUTIONSAMPLE_STACKTRACE(sample), walkState->cache, stackTraceBytes);
+		}
+		freeStackWalkCaches(currentThread, walkState);
+	} else {
+		// TODO: What to do when out of native memory?
```
Comment (tajila) [inline runtime/vm/jfr.cpp:329]: I think we will just need a trace point

**Review by tajila — APPROVED** [review]
LGTM just a couple questions

**Comment by tajila** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk21

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk21

File: runtime/vm/jfr.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/vm/jfr.cpp:64]: Yes, access to the global buffer will need to be synchronized.

File: runtime/vm/jfr.cpp (hunk 2 above)
Comment (gacholio) [reply runtime/vm/jfr.cpp:329]: OK

**Pull request merged**